wallet/src/repository/wallet/mod.rs

```diff
@@ -372,8 +364,24 @@ where
             }
         }
 
+        // Append balance movements of pending blocks
+        state.pending_movements.values().for_each(|x| {
+            keyed_transactions.extend(x.iter().map(|movement| (movement.db_key, movement.clone())));
```

Comment:
The `offset` and `limit` arguments are ignored here, should we check if `movement.db_key` is inside the `range` that the user asked for?

Reply:
Although this is complicated because of `local_movements`, which are always added to the end of the list. So do we would need to subtract `state.local_movements.len()` from the limit when fetching from the database?